src/Client/Updater.php

```diff
@@ -639,12 +636,76 @@ public function download(string $target, ...$extra): PromiseInterface
         }
 
         return $this->repoFileFetcher->fetchTarget($target, $length, ...$extra)
-            ->then(function (StreamInterface $stream) use ($target) {
-                $this->verify($target, $stream);
+            ->then(function (StreamInterface $stream) use ($target, $targetsMetadata) {
+                $this->verify($target, $targetsMetadata, $stream);
                 return $stream;
             });
     }
 
+    /**
+     * Gets a target metadata object that contains the specified target.
+     *
+     * @param string $target
+     *   The path of the target file. Needs to be known to the most recent
+     *   targets metadata downloaded in ::refresh().
+     * @param \Tuf\Metadata\TargetsMetadata|null $targetsMetadata
+     *   The targets metadata to search or null. If null then the search will
+     *   start at the top level 'targets.json' file.
+     * @param string[] $searchedRoles
+     *   The roles that have already been searched.
+     *
+     * @return \Tuf\Metadata\TargetsMetadata|null
+     *   The target metadata with a match for the target, or null no match is
+     *   found.
+     */
+    private function getMetadataForTarget(string $target, ?TargetsMetadata $targetsMetadata = null, array $searchedRoles = []): ?TargetsMetadata
```

Comment:
I think I might need this to be protected, since it will be necessary for the Composer plugin to look up information about specific targets.

Reply:
fixed